File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
         int m = matrix.size();
        if (!m) return;
        int n = matrix[0].size();
        bool first_row_zero = false, first_col_zero = false;
        for (int j = 0; j < n; ++j) if (matrix[0][j] == 0) { first_row_zero = true; break; }
        for (int i = 0; i < m; ++i) if (matrix[i][0] == 0) { first_col_zero = true; break; }
        for (int i = 1; i < m; ++i) {
            auto& row = matrix[i];
            for (int j = 1; j < n; ++j)
                if (row[j] == 0) { row[0] = 0; matrix[0][j] = 0; }
        }
        for (int i = 1; i < m; ++i) {
            auto& row = matrix[i];
            if (row[0] == 0) for (int j = 1; j < n; ++j) row[j] = 0;
        }
        for (int j = 1; j < n; ++j)
            if (matrix[0][j] == 0) for (int i = 1; i < m; ++i) matrix[i][j] = 0;
        if (first_row_zero) for (int j = 0; j < n; ++j) matrix[0][j] = 0;
        if (first_col_zero) for (int i = 0; i < m; ++i) matrix[i][0] = 0;
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        col0 = matrix[0][0]

        for j in range(0,len(matrix[0])):
            if matrix[0][j] == 0:
                col0 = 0
                break

        for i in range(0,len(matrix)):
            if matrix[i][0] == 0:
                matrix[0][0] = 0
                break 

        for j in range(1,len(matrix[0])):
                for i in range(1,len(matrix)):
                    if matrix[i][j] == 0:
                        # matrix[i][j] = 0
                        matrix[0][j] = 0
                        matrix[i][0] = 0

        for j in range(1,len(matrix[0])):
            if matrix[0][j] == 0:
                for i in range(1,len(matrix)):
                    matrix[i][j] = 0

        for i in range(1,len(matrix)):
            if matrix[i][0] == 0:
                for j in range(0,len(matrix[0])):
                    matrix[i][j]=0

        if col0 == 0:
            for j in range(1,len(matrix[0])):
                matrix[0][j]=0


        if matrix[0][0] == 0:
            for i in range(0,len(matrix)):
                matrix[i][0] = 0

        if col0 == 0: 
            matrix[0][0] = 0      

 


        # col = 1
        # for j in range(1,len(matrix[0])):
        #     if matrix[0][j] == 0:
        #         col0  = 0
        #         break

        # if col0 ==0 :
        #     for j in range(1,len(matrix[0])):
        #         matrix[0][j] = 0    


        # row = 1
        # for i in range(0,len(matrix)):
        #     if matrix[i][0] ==0:
        #         row = 0
        #         break 

        # if row == 0 :
        #     for i in range(0,len(matrix)):
        #         matrix[i][0] = 0    
        


            




        